Bottom: 3313c96dcc22cbf1f134c49c35a16540fde9fc11
Top:    d0c81669742823f669ab3e4d84025eba74c5e8d2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-03 22:12:18 +0100

Refresh of hid-gtk-drop-in-gl-renderer-wo

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 0b76c71..ef2d486 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -544,19 +544,23 @@ ghid_invalidate_all ()
 void
 ghid_notify_crosshair_change (bool changes_complete)
 {
+  /* We sometimes get called before the GUI is up */
+  if (gport->drawing_area == NULL)
+    return;
+
   /* FIXME: We could just invalidate the bounds of the crosshair attached objects? */
-  /* FIXME: Seems we're getting prodded before the GUI is up! */
-  if (gport->drawing_area)
-    ghid_invalidate_all ();
+  ghid_invalidate_all ();
 }
 
 void
 ghid_notify_mark_change (bool changes_complete)
 {
+  /* We sometimes get called before the GUI is up */
+  if (gport->drawing_area == NULL)
+    return;
+
   /* FIXME: We could just invalidate the bounds of the mark? */
-  /* FIXME: Seems we're getting prodded before the GUI is up! */
-  if (gport->drawing_area)
-    ghid_invalidate_all ();
+  ghid_invalidate_all ();
 }
 
 static void
